### ISSUE TITLE
test_group_chain fails with python-click 6.7+git20180829-1 

### ISSUE THREAD

**Issue by sebastic** [click-contrib/click-plugins#19]
As reported by Adrian Bunk in [Debian Bug #908684](https://bugs.debian.org/908684), the python-click-plugins package fails to build with python-click 6.7+git20180829-1 due to a test failure:

```
I: pybuild pybuild:269: cp -r /build/python-click-plugins-1.0.2/tests /build/python-click-plugins-1.0.2/.pybuild/cpython2_2.7_click-plugins/build
I: pybuild base:217: cd /build/python-click-plugins-1.0.2/.pybuild/cpython2_2.7_click-plugins/build; python2.7 -m pytest tests
============================= test session starts ==============================
platform linux2 -- Python 2.7.15+, pytest-3.6.4, py-1.6.0, pluggy-0.6.0
rootdir: /build/python-click-plugins-1.0.2, inifile:
collected 5 items

tests/test_plugins.py ...F.                                              [100%]

=================================== FAILURES ===================================
_______________________________ test_group_chain _______________________________

runner = <click.testing.CliRunner object at 0x7f9d888d9050>

    def test_group_chain(runner):
    
        # Attach a sub-group to a CLI and get execute it without arguments to make
        # sure both the sub-group and all the parent group's commands are present
        @good_cli.group()
        def sub_cli():
            """Sub CLI."""
            pass
    
        result = runner.invoke(good_cli)
        assert result.exit_code is 0
        assert sub_cli.name in result.output
        for ep in iter_entry_points('_test_click_plugins.test_plugins'):
            assert ep.name in result.output
    
        # Same as above but the sub-group has plugins
        @with_plugins(plugins=iter_entry_points('_test_click_plugins.test_plugins'))
        @good_cli.group()
        def sub_cli_plugins():
            """Sub CLI with plugins."""
            pass
    
        result = runner.invoke(good_cli, ['sub_cli_plugins'])
>       assert result.exit_code is 0
E       assert 2 is 0
E        +  where 2 = <Result SystemExit(2,)>.exit_code

tests/test_plugins.py:124: AssertionError
====================== 1 failed, 4 passed in 0.12 seconds ======================
E: pybuild pybuild:338: test: plugin distutils failed with: exit code=1: cd /build/python-click-plugins-1.0.2/.pybuild/cpython2_2.7_click-plugins/build; python2.7 -m pytest tests
```

The python-click package was recently updated to a git snapshot which likely changed the behaviour.

**Comment by geowurster** [comment]
Confirmed against https://github.com/pallets/click/commit/035bbfaee74242459ab075e5646e5f4a5574c4a9.  Determining what changed.

**Comment by geowurster** [comment]
https://github.com/pallets/click/commit/5d1df0e042b2f8b0dee8f6dd9ce74edce331a950 changed underscores to dashes in auto-generated CLI command names.  Fixed in #20.  Will push a new release in https://github.com/click-contrib/click-plugins/issues/21.

**Comment by geowurster** [comment]
@sebastic Thanks for catching this.  I pushed a `1.0.4` release to PyPI and a tag.  Shout if things are still broken.

https://github.com/click-contrib/click-plugins/releases/tag/1.0.4

**Comment by sebastic** [comment]
Confirmed fixed in 1.0.4, thanks!
